pennylane/devices/qubit/measure.py

```diff
@@ -194,7 +194,7 @@ def get_measurement_function(
                 return full_dot_products
 
             backprop_mode = math.get_interface(state, *measurementprocess.obs.data) != "numpy"
-            if isinstance(measurementprocess.obs, Hamiltonian):
```

Comment:
We should instead follow the `Sum` logic here instead for the performance improvements. 

Since `CompositeOp` gives it the best effort to compute diagonalizing gates, and benchmarking has shown diagonalizing gates to be substantially more performant when here is no overlapping wires or the total number of wires is less than `7`.  I would assume `LinearCombination` would follow the same performance profile as `Sum`.

Reply:
I am not sure I can follow completely, but with some more explicit instructions I think I could make that fix :)

Reply:
Since `LinearCombination` inherits from `Sum`, removing it from this `isinstance` check should be sufficient.

Reply:
Comment from my PR downstream, removing `LinearCombination` from the `isinstance` check actually causes tests to fail.

Reply:
Yep that is why I restored this after all and noted it to be improved in the LinearCombination upgrade story